discovery_engine_core/core/src/engine.rs

```diff
@@ -813,4 +875,157 @@ mod tests {
     fn test_usize_not_to_small() {
         assert!(size_of::<usize>() >= size_of::<u32>());
     }
+
+    #[tokio::test]
+    async fn test_cross_stack_deduplication() {
+        let mock_server = MockServer::start().await;
+        let tmpl = ResponseTemplate::new(200)
+            .set_body_string(include_str!("../test-fixtures/newscatcher/duplicates.json"));
+
+        Mock::given(method("GET"))
+            .and(path("/_lh"))
+            .respond_with(tmpl)
+            .mount(&mock_server)
+            .await;
+
+        let asset_base = "../../discovery_engine_flutter/example/assets/";
+        let config = InitConfig {
+            api_key: "test-token".to_string(),
+            api_base_url: mock_server.uri(),
+            markets: vec![Market {
+                country_code: "US".to_string(),
+                lang_code: "en".to_string(),
+            }],
+            // This triggers the trusted sources stack to also fetch articles
+            trusted_sources: vec!["example.com".to_string()],
+            excluded_sources: vec![],
+            smbert_vocab: format!("{}/smbert_v0001/vocab.txt", asset_base),
```

Comment:
🎨 we could use the path resolvers in `discovery_engine_core/ai/test-utils/src/smbert.rs` and `discovery_engine_core/ai/test-utils/src/kpe.rs` However we also need add the resolvers for the mocked models because they are currently missing.